Add runInCrossOriginIFrame TestOption for WebKitTestRunner

pascoej opened the pull request:
#### e75e728228a424bc650bc91d343e584c387edaf3
<pre>
Add runInCrossOriginIFrame TestOption for WebKitTestRunner
<a href="https://bugs.webkit.org/show_bug.cgi?id=265984">https://bugs.webkit.org/show_bug.cgi?id=265984</a>
<a href="https://rdar.apple.com/problem/119299992">rdar://problem/119299992</a>

Reviewed by Jonathan Bedard.

This adds a WebKitTestRunner option to run a test in a cross-origin
i-frame. This enables the output of a given test to be compared with the
output of the test in a site isolated cross-origin i-frame by using
run-webkit-tests --self-compare-with-header=&apos;SiteIsolationEnabled=true runInCrossOriginIFrame=true&apos; ...

* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::invoke):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::runInCrossOriginIFrame const):

Canonical link: <a href="https://commits.webkit.org/271718@main">https://commits.webkit.org/271718@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6713f9d51b85babb7a485ec13fcf5961659fde44

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29183 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7852 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30520 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31770 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26542 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/29779 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10001 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5157 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26545 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29455 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6521 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25004 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5616 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5766 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26040 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33107 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26661 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26474 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31983 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5725 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3904 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29768 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7402 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6236 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3783 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6247 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (approved):
We should probably also have an argument to `run-webkit-tests` which transparently enables this.

pascoej committed:
Add runInCrossOriginIFrame TestOption for WebKitTestRunner
https://bugs.webkit.org/show_bug.cgi?id=265984
rdar://problem/119299992

Reviewed by Jonathan Bedard.

This adds a WebKitTestRunner option to run a test in a cross-origin
i-frame. This enables the output of a given test to be compared with the
output of the test in a site isolated cross-origin i-frame by using
run-webkit-tests --self-compare-with-header='SiteIsolationEnabled=true runInCrossOriginIFrame=true' ...

* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::invoke):
* Tools/WebKitTestRunner/TestOptions.cpp:
(WTR::TestOptions::defaults):
(WTR::TestOptions::keyTypeMapping):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::runInCrossOriginIFrame const):

Canonical link: https://commits.webkit.org/271718@main
